spec/features/posts_feature_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'rails_helper'
+
+
+feature 'posts' do
+
+  context 'no posts have been added' do
+
+    scenario 'should display a prompt to add a post' do
+      user = build(:user)
+      visit '/posts'
+      sign_up(user)
+      expect(page).to have_content 'No posts yet'
+      expect(page).to have_link 'Add a post'
+    end
+  end
+
+  context 'posts have been added' do
+
+    scenario 'display posts' do
+      user = build(:user)
+      visit '/posts'
+      sign_up(user)
+      make_post 'hey'
+      expect(page).to have_content('hey')
+      expect(page).not_to have_content('No posts yet')
+    end
+  end
+
+  context 'creating posts' do
+    scenario 'user must be logged in to create a post' do
+      visit '/posts'
+      expect(page).not_to have_content 'Add a post'
+    end
+
+
+  scenario 'prompts user to fill out a form, then displays the new post' do
+    user = build(:user)
+    visit '/posts'
+    sign_up(user)
+    make_post 'test'
+    expect(page).to have_content 'test'
+    expect(current_path).to eq '/posts'
+  end
+end
+
+  context 'an invalid post' do
+    it 'does not let you submit a blank post' do
+      user = build(:user)
+      sign_up(user)
+      click_link 'Add a post'
+      fill_in 'Content', with: ''
+      click_button 'Create Post'
+      expect(page).to have_content 'error'
+    end
+  end
+
+  context 'images' do
+      it 'lets user add images to post' do
+      user = build(:user)
+      sign_up(user)
+      click_link 'Add a post'
+      fill_in 'Content', with: 'test'
+      attach_file 'post_image', "spec/features/files/test.png"
+      click_button 'Create Post'
+      expect(page).to have_xpath("//img[contains(@src, 'test.png')]")
+    end
+  end
+
+  context 'viewing posts' do
+
+  scenario 'lets a user view a post' do
+    user = build(:user)
+    sign_up(user)
+    make_post 'hi there'
+    click_link 'hi there'
+    expect(page).to have_content 'hi there'
+  end
+
+  context 'editing posts' do
+
+  scenario 'let a user edit a post' do
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. (https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)